### DISCUSSION TITLE
restrict a client

### DISCUSSION THREAD

**Discussion by parskaren** [dotnet/MQTTnet#1426] [Q&A]
how can we restrict a client to his/ her related subscribed topics and not show them the other topics?

**Comment by SeppPenner** [comment]
If you're using version 3 or below of the library, check out https://github.com/dotnet/MQTTnet/wiki/Server#intercepting-subscriptions, please. If you're using version 4 already, the question should be forwarded to @chkr1011 as I've not yet switched to the latest version yet.

**Reply by chkr1011** [reply]
For version 4 it is basically the same. The only difference is that there is an event for intercepting subscriptions made by clients. The events args are containing the same properties as the context in version 3.
